### PR TITLE
Feature/create mistake report

### PR DESCRIPTION
**Pull request by ExxZzyy**
Feature/create mistake report, add modal windows

### CODE REVIEW

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts [hunk 1]
```diff
@@ -74,9 +74,16 @@ export class CoachProfileTableComponent implements AfterViewInit, OnInit, OnChan
     }, 10);
   }
 
-  onClick(essayAnswer: string, id: string) {
+  onClick(
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts:77]: maybe we can just pass whole row?

File: src/app/models/coach-profile/coach-profile-dialog/coach-profile-dialog.component.ts [hunk 2]
```diff
@@ -16,20 +16,23 @@ export class CoachProfileDialogComponent implements OnInit {
     private coachCheck: CoachTestsStoreService,
   ) {}
 
+  downloadURL = 'http://elevel-001-site1.btempurl.com/api/File/Download?filePath=';
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-dialog/coach-profile-dialog.component.ts:19]: do you need this

File: src/app/app.module.ts [hunk 3]
```diff
@@ -36,6 +38,8 @@ const INTERCEPTOR_LOADING_INDICATOR: Provider = {
     EditorComponent,
     ManageComponent,
     NotFoundComponent,
+    ReportMistakeDialogComponent,
```
Comment (RomPavel) [inline src/app/app.module.ts:41]: maybe we can define them strictly for common test module

**Review by shuliakleonid — COMMENTED** [review]
I don`t find mistakes

File: src/app/app.module.ts (hunk 3 above)
Comment (ExxZzyy) [reply src/app/app.module.ts]: Done 1abc1a4c8a43c18ce127aa775ea61df54a9d522c

File: src/app/components/report-mistake-dialog/report-mistake-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      auditionId: this.data.auditionId,
+      description: this.description,
+      testId: this.data.testId,
+    };
+    const reportWritingOrSpeaking = {
+      speakingId: this.data.topicId,
+      description: this.description,
+      testId: this.data.testId,
+    };
+    if (this.data.questionId && this.data.auditionId) {
```
Comment (RomPavel) [inline src/app/components/report-mistake-dialog/report-mistake-dialog.component.ts:39]: its not pretty solution. 
its better to have util function, that returns needed submit object depending on this.data, and call createMistakeReport() only once

File: src/app/components/stepper/stepper.component.ts [hunk 5]
```diff
@@ -17,15 +19,19 @@ import { AnswerQuestion, Question } from '../../interfaces/question-answer';
   ],
 })
 export class StepperComponent implements OnChanges, OnInit {
+  rofl: string | undefined;
```
Comment (RomPavel) [inline src/app/components/stepper/stepper.component.ts:22]: funny

File: src/app/app.module.ts (hunk 3 above)
Comment (RomPavel) [reply src/app/app.module.ts]: why you moved only report

Comment (ExxZzyy) [reply src/app/app.module.ts]: Done 8121432b836870ef15e88bc233ea27e132edc395

File: src/app/components/report-mistake-dialog/report-mistake-dialog.component.ts (hunk 4 above)
Comment (ExxZzyy) [reply src/app/components/report-mistake-dialog/report-mistake-dialog.component.ts:32]: Done 8121432b836870ef15e88bc233ea27e132edc395

File: src/app/components/report-mistake-dialog/report-mistake-dialog.component.scss [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  position: relative;
+  width: 30rem;
+}
+
+.container {
+  display: grid;
+}
+
+.mistake__description {
+  margin-bottom: 20px;
```
[the remark this answers is not in the document]
Comment (ExxZzyy) [reply src/app/components/report-mistake-dialog/report-mistake-dialog.component.scss]: Done 8121432b836870ef15e88bc233ea27e132edc395

File: src/app/models/coach-profile/coach-profile-dialog/coach-profile-dialog.component.scss [hunk 7]
```diff
@@ -10,27 +10,25 @@ button.mat-raised-button:not([disabled]):hover {
...
   position: relative;
-  width: 600px;
+  width: 50rem;
 }
 
 .content {
   display: grid;
   grid-template-columns: 70% 30%;
-  padding-top: 50px;
+  padding-top: 40px;
```
[the remark this answers is not in the document]
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-dialog/coach-profile-dialog.component.scss]: Done 8121432b836870ef15e88bc233ea27e132edc395

File: src/app/components/stepper/stepper.component.ts (hunk 5 above)
Comment (ExxZzyy) [reply src/app/components/stepper/stepper.component.ts]: Deleted)  8121432b836870ef15e88bc233ea27e132edc395

**Review by tsiananel — APPROVED** [review]

**Review by shuliakleonid — APPROVED** [review]

**Pull request merged**